finesse/hardware/data_file_writer.py

```diff
@@ -156,7 +164,7 @@ def close(self) -> None:
     def write(self, time: datetime, temperatures: list[Decimal]) -> None:
         """Write temperature readings to the CSV file."""
         # Also include timestamp as seconds since midnight
-        midnight = datetime(time.year, time.month, time.day)
+        midnight = datetime(time.year, time.month, time.day, tzinfo=UTC)
         secs_since_midnight = floor((time - midnight).total_seconds())
```

Comment:
This will produce a warning if `time` does not have timezone information since it is comparing it with a datetime object that do have such information. Where is `time` coming from? Should it be the time taken from the time device, as well?

Reply:
Fair point. `time` comes as a pubsub message from `finesse.hardware._send_temperatures`, which I have set to UTC time there, but maybe some other function will try to use it with a timezone-less `datetime`. Since it's all UTC time anyway, maybe it's best to just use timezone-less datetimes everywhere.